src/modules/sync/advancedDiff.ts

```diff
@@ -0,0 +1,720 @@
+  ContentTypeSnippetModels,
+  TaxonomyModels,
+} from "@kontent-ai/management-sdk";
+import chalk from "chalk";
+import { existsSync, mkdirSync, readFileSync, writeFileSync } from "fs";
+import open from "open";
+import { dirname, resolve } from "path";
+
+import { DiffParams } from "../../commands/diff/diff.js";
+import { logError, logInfo, LogOptions } from "../../log.js";
+import {
+  isCountLimitation,
+  isDefaultValue,
+  isDependsOn,
+  isExternalIdReference,
+  isExternalIdReferenceArray,
+  isMaximumTextLength,
+  isObjectReference,
+  isObjectReferenceArray,
+  isValidationRegex,
+} from "../../utils/typeguards.js";
+import { RequiredCodename } from "../../utils/types.js";
+import { DiffModel, DiffObject } from "./types/diffModel.js";
+import {
+  AddIntoPatchOperation,
+  MovePatchOperation,
+  PatchOperation,
+  RemovePatchOperation,
+  ReplacePatchOperation,
+  ReplacePatchOperationValue,
+} from "./types/patchOperation.js";
+
+type DiffData = DiffModel & Omit<DiffParams, "targetApiKey" | "sourceApiKey">;
+type EntityPathRenderer = {
+  regex: RegExp;
+  render: (() => string) | ((match: string[]) => string);
+};
+type Operation = PatchOperation["op"];
+type TypeOrSnippet = ContentTypeModels.IAddContentTypeData | ContentTypeSnippetModels.IAddContentTypeSnippetData;
+type ElementOrTerm = ContentTypeElements.Element | TaxonomyModels.IAddTaxonomyRequestModel;
+type RenderFunction<T extends PatchOperation> = (patchOp: T) => string;
+type EntityType = "taxonomies" | "types" | "snippets";
+type EntityActionType = "added" | "updated" | "deleted";
+
+// for testing purposes, mocked date passed to spawned process in runCommand()
+const myDate = process.env.MOCKED_DATE ? new Date(process.env.MOCKED_DATE) : new Date();
+const dateGenerated = myDate.toUTCString();
+
+export const generateDiff = (diffData: DiffData) => {
+  const logOptions: LogOptions = diffData;
+  const resolvedPath = resolveOutputPath(diffData.outPath!);
+  const resolvedTemplate = resolveHtmlTemplate("./diffTemplate.html", diffData);
+  const outputDir = dirname(resolvedPath);
+
+  if (!existsSync(outputDir)) {
+    createOutputDirectory(outputDir, logOptions);
+  }
+
+  createOutputFile(resolvedPath, resolvedTemplate, logOptions);
+
+  if (!diffData.noOpen) {
+    openOutputFile(resolvedPath, logOptions);
+  }
+};
+
+const resolveHtmlTemplate = (
+  templatePath: string,
+  diffData: DiffData,
+): string => {
+  const template = readFileSync(resolve(templatePath), "utf-8");
+  const resolvePlaceHolders = (template: string): string => {
+    const processedTemplate = template.replace(
+      /{{.*?}}/g,
+      match => rendererMap.get(match)?.(diffData) ?? match,
+    );
+
+    return processedTemplate === template
+      ? template
+      : resolvePlaceHolders(processedTemplate);
+  };
+
+  return resolvePlaceHolders(template);
+};
+
+const resolveOutputPath = (outputPath: string) => {
+  const hasExtension = !!resolve(outputPath).split("/").pop()?.includes(".");
```

Comment:
why not resolve(outputPath).includes(".")? if you use `resolve` you have absolute path, therefore only files should have '.'

Reply:
smarter, thx.